minio/minio
# Update on-disk storage format for users system

## Description

- Policy mapping is now at `config/iam/policydb/users/myuser1.json`
  and includes version.

- User identity file is now versioned.

- Migrate old data to the new format.

## Motivation and Context

This change is in preparation to add the user-groups feature to MinIO server, so that server admins can create and manage user groups.

In this change, the policy mapped to a user is being moved and on disk json files are being versioned.

Migration from old format to new in included in this change.

## How to test this PR?

Two things to test:

1. Start server on this PR, and check if users system works as before.

2. Start server on master, create some users. Then build this PR and start the server, and make sure users created on master are present in new version on this PR and works as expected.

## Types of changes

## Checklist:
